iroh-net/src/discovery.rs

```diff
@@ -74,6 +75,7 @@ const SOURCE_NAME: &str = "discovery";
 /// refresh, it should start its own task.
 ///
 /// [`RelayUrl`]: crate::relay::RelayUrl
+#[async_trait]
```

Comment:
Can we just use

```rust
fn foo() -> impl Future<...>
```

in the trait (this is allowed now)

and do
```rust
async fn foo() {
```

in the implementation?

I don't see much need for the async_trait macro since RPITIT (return position impl trait in trait) has been stabilized.